rayon-core/src/scope/mod.rs

```diff
@@ -580,24 +579,16 @@ impl<'scope> ScopeBase<'scope> {
             .compare_exchange(nil, &mut *err, Ordering::Release, Ordering::Relaxed)
             .is_ok()
         {
-            log!(JobPanickedErrorStored {
-                owner_thread: self.owner_thread_index
-            });
             mem::forget(err); // ownership now transferred into self.panic
-        } else {
-            log!(JobPanickedErrorNotStored {
-                owner_thread: self.owner_thread_index
-            });
         }
 
-        self.job_completed_latch.set();
+        self.job_completed_latch
+            .set_and_tickle_one(&self.registry, self.owner_thread_index);
     }
 
     unsafe fn job_completed_ok(&self) {
-        log!(JobCompletedOk {
-            owner_thread: self.owner_thread_index
-        });
-        self.job_completed_latch.set();
+        self.job_completed_latch
+            .set_and_tickle_one(&self.registry, self.owner_thread_index);
```

Comment:
I think this is another area where we're potentially invalidating `&self`, when the last spawn sets and its scope races to return. It's not an immediate problem, because these `self.registry` and `self.owner_thread_index` reads are completed as arguments _before_ the actual set. But generally, it's still akin to rust-lang/rust#55005.

Reply:
Is the concern here the (more or less unavoidable, without refactoring to use `*`) UB that rust-lang/rust#55005 describes, or the possibility of accidentally reordering things? It does seem like a subtle invariant that isn't clear from the code.

(Around the latter, I'm wondering if we can refactor in some way to make that less likely... but I don't know how yet.)